src/sonar.ts

```diff
@@ -28,13 +28,70 @@ export async function retrieveSonarCloudIssues(
   return retrieveSonarCloudResults(sonarCloudInputs, url, "issues");
 }
 
+export function buildSonarCloudIssuesUrl({
+  sonarHost,
+  componentKey,
+}: SonarCloudInputs): string {
+  const { prNumber } = getGitHubContext();
+  const path = "/api/issues/search";
+
+  const queryParams = {
+    componentKeys: componentKey,
+    resolved: "false",
+    ps: MAX_PAGE_SIZE,
+    ...(prNumber && { pullRequest: prNumber }),
+  };
+
+  return buildSonarCloudUrl({ sonarHost, path, queryParams });
+}
+
 export async function retrieveSonarCloudHotspots(
   sonarCloudInputs: SonarCloudInputs,
 ): Promise<SonarSearchHotspotResult> {
   const url = buildSonarCloudHotspotsUrl(sonarCloudInputs);
   return retrieveSonarCloudResults(sonarCloudInputs, url, "hotspots");
 }
 
+export function buildSonarCloudHotspotsUrl({
+  sonarHost,
+  componentKey,
+}: SonarCloudInputs): string {
+  const { prNumber } = getGitHubContext();
+  const path = "/api/hotspots/search";
+
+  const queryParams = {
+    projectKey: componentKey,
+    resolved: "false",
+    ps: MAX_PAGE_SIZE,
+    ...(prNumber && { pullRequest: prNumber }),
+  };
+
+  return buildSonarCloudUrl({ sonarHost, path, queryParams });
+}
+
+export function buildSonarCloudUrl({
+  sonarHost,
+  path,
+  queryParams,
+}: {
+  sonarHost: string;
+  path: string;
+  queryParams: { [key: string]: string | number };
+}): string {
+  const baseApiUrl = new URL(sonarHost);
+  const apiUrl = new URL(path, baseApiUrl);
```

Comment:
I think the second parameter can accpet also a String, so maybe you can pass the sonarHost value directly to here instead of createin baseApiUrl